modules/nist_search.py

```diff
@@ -29,69 +29,52 @@ def __str__(self) -> str:
-            if "baseScore" in vuln["impact"]["baseMetricV3"]["cvssV3"]:
-                severity_score = vuln["impact"]["baseMetricV3"]["cvssV3"]["baseScore"]
-
-    elif "baseMetricV2" in vuln["impact"].keys():
-        if "exploitabilityScore" in vuln["impact"]["baseMetricV2"].keys():
-            exploitability = vuln["impact"]["baseMetricV2"]["exploitabilityScore"]
-        elif "cvssV2" in vuln["impact"]["baseMetricV2"]:
-            exploitability = vuln["impact"]["baseMetricV2"]["cvssV2"][
-                "exploitabilityScore"
-            ]
-
-        if "cvssV2" in vuln["impact"]["baseMetricV2"]:
-            if "baseSeverity" in vuln["impact"]["baseMetricV2"]["cvssV2"]:
-                severity = vuln["impact"]["baseMetricV2"]["cvssV2"]["baseSeverity"]
-            elif "severity" in vuln["impact"]["baseMetricV2"].keys():
-                severity = vuln["impact"]["baseMetricV2"]["severity"]
-
-            if "baseScore" in vuln["impact"]["baseMetricV2"]["cvssV2"]:
-                severity_score = vuln["impact"]["baseMetricV2"]["cvssV2"]["baseScore"]
-
-    details_url = vuln["cve"]["references"]["reference_data"][0]["url"]
+    metrics = vuln["cve"].get("metrics")
+    if metrics is not None and len(metrics) > 0:
+        # In testing this appears to contain cvssMetricV31 and cvssMetricV2
+        # Get a list of the score types and sort them in reverse order to get v3 first
+        metrics_types = list(metrics.keys())
+        metrics_types.sort(reverse=True)
+        for score_type in metrics_types:
+            if exploitability == 0.0:
+                exploitability = metrics[score_type][0].get("exploitabilityScore", 0.0)
+            if severity_score == 0.0:
+                severity_score = metrics[score_type][0].get("cvssData", {}).get("baseScore", 0.0)
+            if severity == "UNKNOWN":
+                severity = metrics[score_type][0].get("cvssData", {}).get("baseSeverity", "UNKNOWN")
+
+    details_url = "https://nvd.nist.gov/vuln/detail/" + CVE_ID
 
     return CVE_ID, description, severity, severity_score, details_url, exploitability
 
 
 def searchCVE(keyword: str, log, apiKey=None) -> list[Vulnerability]:
-    url = "https://services.nvd.nist.gov/rest/json/cves/1.0?"
+    url = "https://services.nvd.nist.gov/rest/json/cves/2.0?"
+    # https://services.nvd.nist.gov/rest/json/cves/2.0?keywordSearch=OpenSSH+8.8
     if apiKey:
         sleep_time = 0.1
-        paramaters = {"keyword": keyword, "apiKey": apiKey}
+        headers = {"apiKey": apiKey}
     else:
         sleep_time = 8
-        paramaters = {"keyword": keyword}
+        headers = {}
+    parameters = {"keywordSearch": keyword}
 
     if keyword in cache:
         return cache[keyword]
 
     for tries in range(3):
         try:
             sleep(sleep_time)
-            request = get(url, params=paramaters)
-            data = request.json()
+            response = get(url, params=parameters, headers=headers)
+            data = response.json()
```

Comment:
There still seems to be an issue with this variable.
![image](https://github.com/GamehunterKaan/AutoPWN-Suite/assets/68380968/4ec090d2-2029-4ee1-8440-6a5aa2831194)

Reply:
I added some checks for that variable not existing, tested with `sudo python3 autopwn.py -t 158.174.109.227`

Reply:
please re-review